pybamm/expression_tree/concatenations.py

```diff
@@ -262,17 +269,27 @@ def _concatenation_evaluate(self, children_eval):
         # loop through domains of children writing subvectors to final vector
         for child_vector, slices in zip(children_eval, self._children_slices):
             for child_dom, child_slice in slices.items():
-                vector[self._slices[child_dom]] = child_vector[child_slice]
+                for i, _slice in enumerate(child_slice):
+                    vector[self._slices[child_dom][i]] = child_vector[_slice]
 
         return vector
 
     def _jac(self, variable):
         """ See :meth:`pybamm.Symbol._jac()`. """
-        children = self.cached_children
-        if len(children) == 0:
-            return pybamm.Scalar(0)
-        else:
-            return SparseStack(*[child.jac(variable) for child in children])
+        # note that this assumes that the children are in the right order and only have
```

Comment:
what happens when a child has more than one domain? does this ever happen?

Reply:
currently this doesn't happen in any of the models and `NotImplementedError` would be raised